src/sphmultipcqueue.cpp

```diff
@@ -359,9 +360,11 @@ SPHMPMCQAllocStrideDirectTM (SPHMPMCQ_t queue)
 			if (__TM_is_failure_persistent (TM_buff)) {
 				/* revert to GNU TM */
 #endif
+#ifdef HAS_GNU_TM
 				__transaction_atomic {
 					entryPtr = SPHMPMCQAdvanceHead((SPHLFEntryHeader_t **)&headerBlock->qhead,entrytemp.idUnit,entryfree.idUnit,stride,qlo,qhi);
 				}
+#endif
 #ifdef __HTM__
```

Comment:
No. For the `__HTM__ && HAS_GNU_TM` case, the `HAS_GNU_TM` code is a fallback. If it were an `#elif`, the fallback code would not be present.